apps/judicial-system/web/src/utils/sections.ts

```diff
@@ -18,12 +18,26 @@ import {
   isRulingStepOneValidRC,
   isRulingStepTwoValidIC,
   isRulingStepTwoValidRC,
-} from '../../utils/validate'
+} from '../utils/validate'
+
+interface Section {
+  name: string
+  children: {
+    type: string
+    name: string
+    href: string | undefined
+  }[]
+}
+
+export const findLastValidStep = (section: Section) => {
+  const filterValidSteps = section.children.filter((c) => c.href)
+  return filterValidSteps[filterValidSteps.length - 1]
+}
 
 export const getCustodyAndTravelBanProsecutorSection = (
   workingCase: Case,
   activeSubSection?: number,
-) => {
+): Section => {
```

Comment:
Consider adding  `: Section` to all the other `get...Section` functions in this file